autogen/agentchat/conversable_agent.py

```diff
@@ -970,8 +979,16 @@ def generate_tool_calls_reply(
             function_call = tool_call.get("function", {})
             func = self._function_map.get(function_call.get("name", None), None)
             if inspect.iscoroutinefunction(func):
-                continue
-            _, func_return = self.execute_function(function_call)
+                try:
+                    # get the running loop if it was already created
+                    loop = asyncio.get_running_loop()
+                except RuntimeError:
+                    # create a loop if there is no running loop
+                    loop = asyncio.new_event_loop()
```

Comment:
Does this loop resource get cleaned up once the function call returns?

Reply:
it does now :)

Reply:
😄